File: CommandEventExample/PHP/MyApp/Commands/FooHandler.php
<?php

declare(strict_types=1);


namespace MyApp\Commands;

use MyApp\Events\Fooed;
use MyApp\Events\FooedHandler;
use MyLibrary\Command\Handler;
use MyLibrary\ICommand;

final class FooHandler extends Handler
{

    public function Handle(ICommand $command)
    {
        // Some logic
        echo "Foo command handled\n";

        $result = "{$command->Greeting()}, {$command->Value()}";

        // Obviously we will have an even emiter that will fire the event in real world
        (new FooedHandler())
            ->Handle(new Fooed(new \MyApp\Payloads\Foo($result)));

        return $result;
    }
}

File: Java/CommandsEvents/src/MyApp/Commands/FooHandler.java
package MyApp.Commands;

import MyApp.Events.Fooed;
import MyApp.Events.FooedHandler;
import MyLibrary.Command.Handler;

public final class FooHandler extends Handler<Foo, String>
{
    @Override
    public String Handle(Foo command)
    {
        // Some logic
        System.out.println("Foo command handled");

        String result = String.format("%s, %s", command.Greeting(), command.Value());

        // Obviously we will have an even emitter that will fire the event in real world
        (new FooedHandler())
                .Handle(new Fooed(new MyApp.Payloads.Foo(result)));

        return result;
    }
}
